api/src/main/java/org/eclipse/microprofile/concurrent/ThreadContext.java

```diff
@@ -240,6 +240,44 @@ public static Builder builder() {
      * scopes. An empty/default CDI context means that the thread does not have
      * access to the scope of the session, request, and so forth that created the
      * contextualized action.
+     * 
+     * For example, consider the following <code>&#64;RequestScoped<code> resource:
+     * 
+     * <code><pre>
+    &#64;RequestScoped
+    public class MyRequestScopedBean {
+        public String getState() {
+          // returns some request-specific information to the caller
+        }
+    }
+    </pre></code>
+     * 
+     * When CDI context is propagated, the request's state will also
+     * be available to the contextualized work: <code><pre>
+    ManagedExecutor exec = ManagedExecutor.builder()
+        .propagated(ThreadContext.CDI, ThreadContext.APPLICATION)
+        .build();
+    
+    &#64;Inject
+    MyRequestScopedBean requestBean;
+    
+    &#64;GET
+    public void foo() {
+        exec.supplyAsync(() -> {
+            String state = reqBean.getState();
+            // do some work with the request state
+        }).thenApply(more -> {
+            // request state also available in future stages
+        });
+    }
+    </pre></code>
+     *
+     * If CDI context is 'cleared', the state of any Request, Session, or Conversation
+     * scoped beans will be cleared during task execution, and restored after execution.
+     * 
+     * If CDI context is 'unchanged', access to CDI bean's contextual state 
```

Comment:
Hmm, this is somewhat tricky. I'd almost lean towards saying it's bad practice to leave it `unchanged` or otherwise discourage users from going down this path _if they want to use any CDI beans_.
Because based on what thread runs it, you may be able to access beans (if on main thread) or you will hit a wall head on with `ContextNotActiveException` (on any new thread).